composeApp/src/androidMain/kotlin/org/ooni/probe/MainActivity.kt

```diff
@@ -50,18 +51,29 @@ class MainActivity : ComponentActivity() {
     }
 
     private fun manageIntent(intent: Intent) {
-        if (intent.action != Intent.ACTION_VIEW) return
-        val uri = intent.data ?: return
-        when (uri.host) {
-            "runv2",
-            OrganizationConfig.ooniRunDomain,
-            -> {
-                val id = uri.lastPathSegment ?: return
-                deepLinkFlow.tryEmit(DeepLink.AddDescriptor(id))
-            }
+        if (intent.action != Intent.ACTION_VIEW && intent.action != Intent.ACTION_SEND) return
+
+        if (intent.action == Intent.ACTION_VIEW) {
+            val uri = intent.data ?: return
+            when (uri.host) {
+                "runv2",
+                OrganizationConfig.ooniRunDomain,
+                -> {
+                    val id = uri.lastPathSegment ?: return
+                    deepLinkFlow.tryEmit(DeepLink.AddDescriptor(id))
+                }
 
-            else -> {
-                Logger.e { "Unknown deep link: $uri" }
+                else -> {
+                    Logger.e { "Unknown deep link: $uri" }
+                }
+            }
+        } else if (intent.action == Intent.ACTION_SEND) {
+            val url = intent.getStringExtra(Intent.EXTRA_TEXT) ?: return
+            if (Patterns.WEB_URL.matcher(url).matches()) {
+                deepLinkFlow.tryEmit(DeepLink.RunUrls(url))
+            } else {
+                Logger.e { "Unknown deep link: $url" }
+                return
             }
```

Comment:
Unlike the deeplink, I think this case needs a bit more feedback to the user. We can't filter that the intents we get are URLs. That means the OONI app will appear in a lot more Share dialogs. And users won't know what the app supports. So, even if the users shares an invalid URL (a piece of text), we should open the ChooseWebsites screen and show an error explaining That OONI only supports sharing URLs to test. What do you think?

Also, we need to be careful with Logger.e or Logger.w. They always generate a non-crashing error on Sentry.

Reply:
The same should probably also apply to the iOS app.